src/pkg/services/m365/api/users.go

```diff
@@ -283,6 +284,22 @@ func (c Users) getMailboxSettings(
 	return settings, nil
 }
 
+func (c Users) IsExchangeServiceEnabled(
```

Comment:
This is effectively a duplicate of the service-layer [UserHasMailbox()](https://github.com/alcionai/corso/blob/main/src/pkg/services/m365/users.go#L63) func.  Can we use that func instead of adding a new one?

As a design philosophy: more interperative functionality like this (ie, calling some api and transforming the results according to a given interpretation) are better located in the service layer, not the api layer.  The API layer is an owner for each specific calls itself.  Not wrappers on those calls.  We're not perfect on this, but that's the ideal.

That would ultimately mean we'd want to move GetMailInbox to the service layer.  Granted, that can feel a little weird when corso itself wants to consume the func, since the service layer is SDK oriented.  Hmm.  Maybe the best middle ground is to remove this func, keep with UserHasMailbox in the service layer, have corso call GetMailInbox, and better integrate EvaluateMailboxError into GetMailInbox.  That would set us up to remove this intermediate func.

Reply:
> . Granted, that can feel a little weird when corso itself wants to consume the func, since the service layer is SDK oriented.

Yeah, this was my thinking behind adding this func. I want common code which checks if exchange is enabled. Currently we do this separately for SDK & CLI which can lead to bugs.

> and better integrate EvaluateMailboxError into GetMailInbox.

Let me look into it.